packages/concerto-core/lib/serializer/objectvalidator.js

```diff
@@ -120,6 +120,15 @@ class ObjectValidator {
 
         const obj = parameters.stack.pop();
 
+        if (!obj.$class){
+            // Choose the only concrete subclass if it exists
+            const subclasses = classDeclaration.getAssignableClassDeclarations()
+                .filter(clazz => !clazz.isAbstract());
```

Comment:
Does the getAssignableClassDeclarations method also return the base type? If so, the comment should probably be corrected to say "class" instead of "subclass". If not, I think the code should also handle the more common case where the actual type is equal to the declared type, and not a subclass of it.

Reply:
Thanks Adam. `getAssignableClassDeclarations` does return itself too, so I've updated the comments to reflect this.